### DISCUSSION TITLE
When typing the selection keeps expanding.

### DISCUSSION THREAD

**Discussion by MGlolenstine** [helix-editor/helix#2847] [Q&A] [answered]
When I enter an insert mode from a selection and start typing, the text I typed gets selected (highlighted).

Is that intended?

**Comment by the-mikedavis** [comment, marked as answer]
Yes this is expected behavior. It's a little visually distracting at first but you get used to it and it's a powerful tool when you're working with multiple selections.

**Reply by MGlolenstine** [reply]
Could I get some deeper insight into how that selection is useful?
(I'm not saying it's not, but I'm willing to learn, and I'd love to see some use cases)

**Reply by the-mikedavis** [reply]
If you haven't already, I would recommend checking out the tutor (`hx --tutor` or `:tutor` within helix). The [kakoune documentation](http://kakoune.org/) is also very good. Kakoune was the inspiration for multiple selections and the select-then-act paradigm

**Reply by MGlolenstine** [reply]
Thank you!
